docs/fundamentals/snippets/healthz/Healthz.ServiceDefaults/Extensions.cs

```diff
@@ -0,0 +1,59 @@
+﻿using Microsoft.AspNetCore.Builder;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+
+namespace Microsoft.Extensions.Hosting;
+
+public static class Extensions
+{
+    public static IHostApplicationBuilder AddServiceDefaults(this IHostApplicationBuilder builder)
+    {
+        builder.AddDefaultHealthChecks();
+
+        return builder;
+    }
+
+    // <healthchecks>
+    public static IHostApplicationBuilder AddDefaultHealthChecks(this IHostApplicationBuilder builder)
+    {
+        builder.Services.AddRequestTimeouts(
+            configure: static timeouts =>
+                timeouts.AddPolicy("HealthChecks", TimeSpan.FromSeconds(5)));
+
+        builder.Services.AddOutputCache(
+            configureOptions: static caching =>
+                caching.AddPolicy("HealthChecks",
+                build: static policy => policy.Expire(TimeSpan.FromSeconds(10))));
+
+        builder.Services.AddHealthChecks()
+            // Add a default liveness check to ensure app is responsive
+            .AddCheck("self", () => HealthCheckResult.Healthy(), ["live"]);
+
+        return builder;
+    }
+    // </healthchecks>
+
+    // <mapendpoints>
+    public static WebApplication MapDefaultEndpoints(this WebApplication app)
+    {
+        var healthChecks = app.MapGroup("");
+
+        healthChecks
+            .CacheOutput("HealthChecks")
+            .WithRequestTimeout("HealthChecks");
+
+        // All health checks must pass for app to be
+        // considered ready to accept traffic after starting
+        healthChecks.MapHealthChecks("/health");
```

Comment:
Is there a reason this is called "healthz" but the health endpoing is `health`?

Reply:
The folder and the project name, which is not visible on docs, are both named healthz - for an internal reminder that it was a named that was considered before. The endpoint is `health` since that's accurate.